src/exif.c

```diff
@@ -142,9 +142,10 @@ avifResult avifImageExtractExifOrientationToIrotImir(avifImage * image)
     return AVIF_RESULT_OK;
 }
 
-void avifImageSetMetadataExif(avifImage * image, const uint8_t * exif, size_t exifSize)
+avifResult avifImageSetMetadataExif(avifImage * image, const uint8_t * exif, size_t exifSize)
 {
-    avifRWDataSet(&image->exif, exif, exifSize);
+    AVIF_CHECKRES(avifRWDataSet(&image->exif, exif, exifSize));
     // Ignore any Exif parsing failure.
     (void)avifImageExtractExifOrientationToIrotImir(image);
+    return AVIF_RESULT_OK;
```

Comment:
Not sure what to do here.
On one hand, libavif already reports Exif parsing failures so `avifImageSetMetadataExif()` does not have to. `avifImageSetMetadataExif()` can be used in other contexts than for AVIF encoding, and keeping users from using it with a malformed Exif chunk could be limiting.
On the other hand, the user can still use `avifImageSetMetadataExif()` and ignore the Exif parsing error. This needs proper documentation though.

This is a minor question so I am leaning towards leaving it as is and saving time.

Reply:
It is fine to add a TODO(wtc) comment so that the code review can conclude quickly.

One reason we ignore the failure of `avifImageExtractExifOrientationToIrotImir()` here is that `avifImageSetMetadataExif()` returns void. The TODO(wtc) comment in tests/gtest/avifmetadatatest.cc removed in this pull request is also about this issue.

Reply:
Keep the original comment `// Ignore any Exif parsing failure.`

Add the TODO comment after the original comment.